pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix

```diff
@@ -40,6 +40,15 @@ let
         let
           grammars = map grammarToPlugin
             (f (tree-sitter.builtGrammars // builtGrammars));
+          filterDuplicateGrammars = lib.lists.foldl' (
+            res: new-el:
+              if lib.lists.any (old-el:
+                # `meta.name` contains something like `c-grammar-0.0.0+rev=be23d2c`
+                new-el.meta.name == old-el.meta.name
+              ) res
+              then res
```

Comment:
Should we log the drop here, e.g. `(lib.warn "ignoring duplicate grammar ${lib.getName res}" res)` or is that too noisy? If we warn, then we could have _both_ this filtering mechanism (which provides the warnings) _and_ `ln -sf` (which guarantees the build always succeeds).

Reply:
unfortunately, too noisy... if you have duplicates its highly possible to have the entire set duplicated, not just a couple.

Reply:
and if you do have duplicates, you cant DO a ton about it

Reply:
if you were to add a warning, it would have to be AFTER filtering, and done in bash.

So, if we did ln -sf, we could do a quick [ -e $file ] && echo "duplicates!!!" warning to at least let people know it unpredictably skipped one.

but warning on every item we filter is a no go

Reply:
sounds good. I'll tentatively approve then, and defer to you for additional comments.

Reply:
for reference, heres what it looks like when you trace the removed ones hahaha

```
ix build --show-trace .
warning: Git tree '/home/birdee/birdeeSystems' is dirty
trace: vimplugin-treesitter-grammar-c_sharp
trace: vimplugin-treesitter-grammar-embedded_template
trace: vimplugin-treesitter-grammar-git_config
trace: vimplugin-treesitter-grammar-git_rebase
trace: vimplugin-treesitter-grammar-godot_resource
trace: vimplugin-treesitter-grammar-haskell_persistent
trace: vimplugin-treesitter-grammar-janet_simple
trace: vimplugin-treesitter-grammar-markdown_inline
trace: vimplugin-treesitter-grammar-nim_format_string
trace: vimplugin-treesitter-grammar-ocaml_interface
trace: vimplugin-treesitter-grammar-php_only
trace: vimplugin-treesitter-grammar-poe_filter
trace: vimplugin-treesitter-grammar-ssh_config
trace: vimplugin-treesitter-grammar-ada
trace: vimplugin-treesitter-grammar-agda
trace: vimplugin-treesitter-grammar-angular
trace: vimplugin-treesitter-grammar-apex
trace: vimplugin-treesitter-grammar-arduino
trace: vimplugin-treesitter-grammar-asm
trace: vimplugin-treesitter-grammar-astro
trace: vimplugin-treesitter-grammar-authzed
trace: vimplugin-treesitter-grammar-awk
trace: vimplugin-treesitter-grammar-bash
trace: vimplugin-treesitter-grammar-bass
trace: vimplugin-treesitter-grammar-beancount
trace: vimplugin-treesitter-grammar-bibtex
trace: vimplugin-treesitter-grammar-bicep
trace: vimplugin-treesitter-grammar-bitbake
trace: vimplugin-treesitter-grammar-blueprint
trace: vimplugin-treesitter-grammar-bp
trace: vimplugin-treesitter-grammar-c
trace: vimplugin-treesitter-grammar-c_sharp
trace: vimplugin-treesitter-grammar-c_sharp
trace: vimplugin-treesitter-grammar-cairo
trace: vimplugin-treesitter-grammar-capnp
trace: vimplugin-treesitter-grammar-chatito
trace: vimplugin-treesitter-grammar-clojure
trace: vimplugin-treesitter-grammar-cmake
trace: vimplugin-treesitter-grammar-comment
trace: vimplugin-treesitter-grammar-commonlisp
trace: vimplugin-treesitter-grammar-cooklang
trace: vimplugin-treesitter-grammar-corn
trace: vimplugin-treesitter-grammar-cpon
trace: vimplugin-treesitter-grammar-cpp
trace: vimplugin-treesitter-grammar-css
trace: vimplugin-treesitter-grammar-csv
trace: vimplugin-treesitter-grammar-cuda
trace: vimplugin-treesitter-grammar-cue
trace: vimplugin-treesitter-grammar-d
trace: vimplugin-treesitter-grammar-dart
trace: vimplugin-treesitter-grammar-devicetree
trace: vimplugin-treesitter-grammar-dhall
trace: vimplugin-treesitter-grammar-diff
trace: vimplugin-treesitter-grammar-disassembly
trace: vimplugin-treesitter-grammar-djot
trace: vimplugin-treesitter-grammar-dockerfile
trace: vimplugin-treesitter-grammar-dot
trace: vimplugin-treesitter-grammar-doxygen
trace: vimplugin-treesitter-grammar-dtd
trace: vimplugin-treesitter-grammar-earthfile
trace: vimplugin-treesitter-grammar-ebnf
trace: vimplugin-treesitter-grammar-editorconfig
trace: vimplugin-treesitter-grammar-eds
trace: vimplugin-treesitter-grammar-eex
trace: vimplugin-treesitter-grammar-elixir
trace: vimplugin-treesitter-grammar-elm
trace: vimplugin-treesitter-grammar-elsa
trace: vimplugin-treesitter-grammar-elvish
trace: vimplugin-treesitter-grammar-embedded_template
trace: vimplugin-treesitter-grammar-embedded_template
trace: vimplugin-treesitter-grammar-erlang
trace: vimplugin-treesitter-grammar-facility
trace: vimplugin-treesitter-grammar-faust
trace: vimplugin-treesitter-grammar-fennel
trace: vimplugin-treesitter-grammar-fidl
trace: vimplugin-treesitter-grammar-firrtl
trace: vimplugin-treesitter-grammar-fish
trace: vimplugin-treesitter-grammar-foam
trace: vimplugin-treesitter-grammar-forth
trace: vimplugin-treesitter-grammar-fortran
trace: vimplugin-treesitter-grammar-fsh
trace: vimplugin-treesitter-grammar-func
trace: vimplugin-treesitter-grammar-fusion
trace: vimplugin-treesitter-grammar-gdscript
trace: vimplugin-treesitter-grammar-gdshader
trace: vimplugin-treesitter-grammar-git_config
trace: vimplugin-treesitter-grammar-git_rebase
trace: vimplugin-treesitter-grammar-git_config
trace: vimplugin-treesitter-grammar-git_rebase
trace: vimplugin-treesitter-grammar-gitattributes
trace: vimplugin-treesitter-grammar-gitcommit
trace: vimplugin-treesitter-grammar-gitignore
trace: vimplugin-treesitter-grammar-gleam
trace: vimplugin-treesitter-grammar-glimmer
trace: vimplugin-treesitter-grammar-glsl
trace: vimplugin-treesitter-grammar-gn
trace: vimplugin-treesitter-grammar-gnuplot
trace: vimplugin-treesitter-grammar-go
trace: vimplugin-treesitter-grammar-goctl
trace: vimplugin-treesitter-grammar-godot_resource
trace: vimplugin-treesitter-grammar-godot_resource
trace: vimplugin-treesitter-grammar-gomod
trace: vimplugin-treesitter-grammar-gosum
trace: vimplugin-treesitter-grammar-gotmpl
trace: vimplugin-treesitter-grammar-gowork
trace: vimplugin-treesitter-grammar-gpg
trace: vimplugin-treesitter-grammar-graphql
trace: vimplugin-treesitter-grammar-groovy
trace: vimplugin-treesitter-grammar-gstlaunch
trace: vimplugin-treesitter-grammar-hack
trace: vimplugin-treesitter-grammar-hare
trace: vimplugin-treesitter-grammar-haskell
trace: vimplugin-treesitter-grammar-haskell_persistent
trace: vimplugin-treesitter-grammar-haskell_persistent
trace: vimplugin-treesitter-grammar-hcl
trace: vimplugin-treesitter-grammar-heex
trace: vimplugin-treesitter-grammar-helm
trace: vimplugin-treesitter-grammar-hjson
trace: vimplugin-treesitter-grammar-hlsl
trace: vimplugin-treesitter-grammar-hlsplaylist
trace: vimplugin-treesitter-grammar-hocon
trace: vimplugin-treesitter-grammar-hoon
trace: vimplugin-treesitter-grammar-html
trace: vimplugin-treesitter-grammar-htmldjango
trace: vimplugin-treesitter-grammar-http
trace: vimplugin-treesitter-grammar-hurl
trace: vimplugin-treesitter-grammar-hyprlang
trace: vimplugin-treesitter-grammar-idl
trace: vimplugin-treesitter-grammar-ini
trace: vimplugin-treesitter-grammar-inko
trace: vimplugin-treesitter-grammar-ispc
trace: vimplugin-treesitter-grammar-janet_simple
trace: vimplugin-treesitter-grammar-janet_simple
trace: vimplugin-treesitter-grammar-java
trace: vimplugin-treesitter-grammar-javascript
trace: vimplugin-treesitter-grammar-jq
trace: vimplugin-treesitter-grammar-jsdoc
trace: vimplugin-treesitter-grammar-json
trace: vimplugin-treesitter-grammar-json5
trace: vimplugin-treesitter-grammar-jsonc
trace: vimplugin-treesitter-grammar-jsonnet
trace: vimplugin-treesitter-grammar-julia
trace: vimplugin-treesitter-grammar-just
trace: vimplugin-treesitter-grammar-kconfig
trace: vimplugin-treesitter-grammar-kdl
trace: vimplugin-treesitter-grammar-kotlin
trace: vimplugin-treesitter-grammar-koto
trace: vimplugin-treesitter-grammar-kusto
trace: vimplugin-treesitter-grammar-lalrpop
trace: vimplugin-treesitter-grammar-latex
trace: vimplugin-treesitter-grammar-ledger
trace: vimplugin-treesitter-grammar-leo
trace: vimplugin-treesitter-grammar-linkerscript
trace: vimplugin-treesitter-grammar-liquid
trace: vimplugin-treesitter-grammar-liquidsoap
trace: vimplugin-treesitter-grammar-llvm
trace: vimplugin-treesitter-grammar-lua
trace: vimplugin-treesitter-grammar-luadoc
trace: vimplugin-treesitter-grammar-luap
trace: vimplugin-treesitter-grammar-luau
trace: vimplugin-treesitter-grammar-m68k
trace: vimplugin-treesitter-grammar-make
trace: vimplugin-treesitter-grammar-markdown
trace: vimplugin-treesitter-grammar-markdown_inline
trace: vimplugin-treesitter-grammar-markdown_inline
trace: vimplugin-treesitter-grammar-matlab
trace: vimplugin-treesitter-grammar-menhir
trace: vimplugin-treesitter-grammar-mermaid
trace: vimplugin-treesitter-grammar-meson
trace: vimplugin-treesitter-grammar-mlir
trace: vimplugin-treesitter-grammar-muttrc
trace: vimplugin-treesitter-grammar-nasm
trace: vimplugin-treesitter-grammar-nginx
trace: vimplugin-treesitter-grammar-nickel
trace: vimplugin-treesitter-grammar-nim
trace: vimplugin-treesitter-grammar-nim_format_string
trace: vimplugin-treesitter-grammar-nim_format_string
trace: vimplugin-treesitter-grammar-ninja
trace: vimplugin-treesitter-grammar-nix
trace: vimplugin-treesitter-grammar-norg
trace: vimplugin-treesitter-grammar-nqc
trace: vimplugin-treesitter-grammar-objc
trace: vimplugin-treesitter-grammar-objdump
trace: vimplugin-treesitter-grammar-ocaml
trace: vimplugin-treesitter-grammar-ocaml_interface
trace: vimplugin-treesitter-grammar-ocaml_interface
trace: vimplugin-treesitter-grammar-ocamllex
trace: vimplugin-treesitter-grammar-odin
trace: vimplugin-treesitter-grammar-org
trace: vimplugin-treesitter-grammar-pascal
trace: vimplugin-treesitter-grammar-passwd
trace: vimplugin-treesitter-grammar-pem
trace: vimplugin-treesitter-grammar-perl
trace: vimplugin-treesitter-grammar-php
trace: vimplugin-treesitter-grammar-php_only
trace: vimplugin-treesitter-grammar-php_only
trace: vimplugin-treesitter-grammar-phpdoc
trace: vimplugin-treesitter-grammar-pioasm
trace: vimplugin-treesitter-grammar-po
trace: vimplugin-treesitter-grammar-pod
trace: vimplugin-treesitter-grammar-poe_filter
trace: vimplugin-treesitter-grammar-poe_filter
trace: vimplugin-treesitter-grammar-pony
trace: vimplugin-treesitter-grammar-powershell
trace: vimplugin-treesitter-grammar-printf
trace: vimplugin-treesitter-grammar-prisma
trace: vimplugin-treesitter-grammar-problog
trace: vimplugin-treesitter-grammar-prolog
trace: vimplugin-treesitter-grammar-promql
trace: vimplugin-treesitter-grammar-properties
trace: vimplugin-treesitter-grammar-proto
trace: vimplugin-treesitter-grammar-prql
trace: vimplugin-treesitter-grammar-psv
trace: vimplugin-treesitter-grammar-pug
trace: vimplugin-treesitter-grammar-puppet
trace: vimplugin-treesitter-grammar-purescript
trace: vimplugin-treesitter-grammar-pymanifest
trace: vimplugin-treesitter-grammar-python
trace: vimplugin-treesitter-grammar-ql
trace: vimplugin-treesitter-grammar-qmldir
trace: vimplugin-treesitter-grammar-qmljs
trace: vimplugin-treesitter-grammar-query
trace: vimplugin-treesitter-grammar-r
trace: vimplugin-treesitter-grammar-racket
trace: vimplugin-treesitter-grammar-ralph
trace: vimplugin-treesitter-grammar-rasi
trace: vimplugin-treesitter-grammar-rbs
trace: vimplugin-treesitter-grammar-re2c
trace: vimplugin-treesitter-grammar-readline
trace: vimplugin-treesitter-grammar-regex
trace: vimplugin-treesitter-grammar-rego
trace: vimplugin-treesitter-grammar-requirements
trace: vimplugin-treesitter-grammar-rescript
trace: vimplugin-treesitter-grammar-rnoweb
trace: vimplugin-treesitter-grammar-robot
trace: vimplugin-treesitter-grammar-robots
trace: vimplugin-treesitter-grammar-roc
trace: vimplugin-treesitter-grammar-ron
trace: vimplugin-treesitter-grammar-rst
trace: vimplugin-treesitter-grammar-ruby
trace: vimplugin-treesitter-grammar-rust
trace: vimplugin-treesitter-grammar-scala
trace: vimplugin-treesitter-grammar-scfg
trace: vimplugin-treesitter-grammar-scheme
trace: vimplugin-treesitter-grammar-scss
trace: vimplugin-treesitter-grammar-slang
trace: vimplugin-treesitter-grammar-slint
trace: vimplugin-treesitter-grammar-smali
trace: vimplugin-treesitter-grammar-smithy
trace: vimplugin-treesitter-grammar-snakemake
trace: vimplugin-treesitter-grammar-solidity
trace: vimplugin-treesitter-grammar-soql
trace: vimplugin-treesitter-grammar-sosl
trace: vimplugin-treesitter-grammar-sourcepawn
trace: vimplugin-treesitter-grammar-sparql
trace: vimplugin-treesitter-grammar-sql
trace: vimplugin-treesitter-grammar-squirrel
trace: vimplugin-treesitter-grammar-ssh_config
trace: vimplugin-treesitter-grammar-ssh_config
trace: vimplugin-treesitter-grammar-starlark
trace: vimplugin-treesitter-grammar-strace
trace: vimplugin-treesitter-grammar-styled
trace: vimplugin-treesitter-grammar-supercollider
trace: vimplugin-treesitter-grammar-surface
trace: vimplugin-treesitter-grammar-svelte
trace: vimplugin-treesitter-grammar-swift
trace: vimplugin-treesitter-grammar-sxhkdrc
trace: vimplugin-treesitter-grammar-systemtap
trace: vimplugin-treesitter-grammar-systemverilog
trace: vimplugin-treesitter-grammar-t32
trace: vimplugin-treesitter-grammar-tablegen
trace: vimplugin-treesitter-grammar-tact
trace: vimplugin-treesitter-grammar-tcl
trace: vimplugin-treesitter-grammar-teal
trace: vimplugin-treesitter-grammar-templ
trace: vimplugin-treesitter-grammar-terraform
trace: vimplugin-treesitter-grammar-textproto
trace: vimplugin-treesitter-grammar-thrift
trace: vimplugin-treesitter-grammar-tiger
trace: vimplugin-treesitter-grammar-tlaplus
trace: vimplugin-treesitter-grammar-tmux
trace: vimplugin-treesitter-grammar-todotxt
trace: vimplugin-treesitter-grammar-toml
trace: vimplugin-treesitter-grammar-wgsl_bevy
trace: vimplugin-treesitter-grammar-tsv
trace: vimplugin-treesitter-grammar-tsx
trace: vimplugin-treesitter-grammar-turtle
trace: vimplugin-treesitter-grammar-twig
trace: vimplugin-treesitter-grammar-typescript
trace: vimplugin-treesitter-grammar-typespec
trace: vimplugin-treesitter-grammar-typoscript
trace: vimplugin-treesitter-grammar-typst
trace: vimplugin-treesitter-grammar-udev
trace: vimplugin-treesitter-grammar-ungrammar
trace: vimplugin-treesitter-grammar-unison
trace: vimplugin-treesitter-grammar-usd
trace: vimplugin-treesitter-grammar-uxntal
trace: vimplugin-treesitter-grammar-v
trace: vimplugin-treesitter-grammar-vala
trace: vimplugin-treesitter-grammar-vento
trace: vimplugin-treesitter-grammar-verilog
trace: vimplugin-treesitter-grammar-vhdl
trace: vimplugin-treesitter-grammar-vhs
trace: vimplugin-treesitter-grammar-vim
trace: vimplugin-treesitter-grammar-vimdoc
trace: vimplugin-treesitter-grammar-vrl
trace: vimplugin-treesitter-grammar-vue
trace: vimplugin-treesitter-grammar-wgsl
trace: vimplugin-treesitter-grammar-wgsl_bevy
trace: vimplugin-treesitter-grammar-wgsl_bevy
trace: vimplugin-treesitter-grammar-wing
trace: vimplugin-treesitter-grammar-wit
trace: vimplugin-treesitter-grammar-xcompose
trace: vimplugin-treesitter-grammar-xml
trace: vimplugin-treesitter-grammar-yaml
trace: vimplugin-treesitter-grammar-yang
trace: vimplugin-treesitter-grammar-yuck
trace: vimplugin-treesitter-grammar-zathurarc
trace: vimplugin-treesitter-grammar-zig
```